### PR TITLE
Add support for filtering k8s endpoints by port

### LINKED ISSUE #370: Add support for filtering k8s endpoints by port
Hi,

We are utilizing RDS to automatically discover k8s endpoints for probing. One minor issue we are running into when services include multiple ports for different protocols, for example 80/443 for http/https. In this case it would be desirable to have a http probe for port 80, and a https probe for port 443. But since both ports are discovered together, any probe will target both of them, leading to things like:
```
Target:example.com, URL:https://10.x.x.x:80/ping, http.doHTTPRequest: Get https://10.x.x.x:80/ping: http: server gave HTTP response to HTTPS client
```

A couple workarounds are available:
1. Explicitly set the port in each probe to override the discovered endpoint values. In this example the http probe would specify 80, and the https probe would specify 443. The downside of this is it needlessly doubles the number of probes.
1. Setup additional headless services selecting the same pods which each expose the single port we want to probe. The downside of this is extra overhead for the k8s cluster, and the added maintenance costs of keeping the services in sync.

To address those shortcomings, I was thinking about extending the filter mechanism and adding one for port. It could easily be applied to the resources returned inside the loop inside `kubernetes.(*epLister).listResources`. If that sounds reasonable, I'd be happy to submit a PR. Thanks!

### PR DESCRIPTION
**Pull request by cbroglie**
Fixes #370

### CODE REVIEW

**Comment by cbroglie** [comment]
@googlebot I signed it!

**Comment by cbroglie** [comment]
The test failure is unrelated, looks like a flappy test.

File: rds/kubernetes/endpoints.go [hunk 1]
```diff
@@ -111,7 +111,9 @@ type epInfo struct {
 // is composed of multiple addresses and ports. If an endpoint subset as 3
 // addresses and 2 ports, there will be 6 resources corresponding to that
 // subset.
-func (epi *epInfo) resources() (resources []*pb.Resource) {
+func (epi *epInfo) resources(allFilters *filter.Filters, l *logger.Logger) (resources []*pb.Resource) {
```
Comment (manugarg) [inline rds/kubernetes/endpoints.go]: Instead of passing all filters, I'd just pass portFilter to this method.

File: rds/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -72,7 +76,7 @@ var SupportedFilters = struct {
 	RegexFilterKeys []string
 	LabelsFilter    bool
 }{
-	[]string{"name", "namespace"},
+	[]string{"name", "namespace", "port"},
```
Comment (manugarg) [inline rds/kubernetes/kubernetes.go]: May be add a comment here saying that "port" filter applies only to endpoints.

**Comment by cbroglie** [comment]
Thanks! Updated to address the comments.

**Comment by manugarg** [comment]
LGTM. I've started the commit process described at:
https://github.com/google/cloudprober/blob/master/CONTRIBUTING.md#source-of-truth-sot-and-commit-process

**Comment by cbroglie** [comment]
Thanks!

**Comment by manugarg** [comment]
This change has been committed through: #377.

Once again, thanks for the contribution. 👍

**Pull request closed without merging**